language/move-vm/types/src/gas_schedule.rs

```diff
@@ -93,6 +93,8 @@ impl<'a> CostStrategy<'a> {
         opcode: Opcodes,
         size: AbstractMemorySize<GasCarrier>,
     ) -> PartialVMResult<()> {
+        // Make sure that the size is always non-zero
```

Comment:
In `deduct_gas` itself, I think we should add some logging/debug_assert for if `amount` is zero. Thoughts?

Reply:
Sadly this will not work as we use the `zero_cost_schedule` in places (genesis, writeset prologue/epilogue, writeset execution, and block prologue) and that will violate that the number of gas units will always be non-zero

Reply:
maybe we could make 2 API entry points? or would that be too tricky?

Reply:
Maybe at the very least for `charge_instr`?

Reply:
I think making two API entrypoints doesn't work really here: it's running normal Move code through the interpreter but with a zero cost schedule (i.e., the interpreter loop is parametrized in-part by the gas schedule), so in order to do this we would need to trait-ify the gas schedule and have a zero cost implementation of this for that trait which seems overly-complex.

Reply:
Let's schedule some time to talk about this. I'm a bit confused about the limitations after reading the code